test/docker-e2e/docker_test.go

```diff
@@ -0,0 +1,270 @@
+func (s *DockerTestSuite) SetupDockerResources(opts ...ConfigOption) {
+	s.provider = s.CreateDockerProvider(opts...)
+	s.celestia = s.CreateChain()
+	s.daNetwork = s.CreateDANetwork()
+	s.rollkitChain = s.CreateRollkitChain()
+}
+
+// CreateChain creates a chain using the provider.
+func (s *DockerTestSuite) CreateChain() tastoratypes.Chain {
+	ctx := context.Background()
+
+	chain, err := s.provider.GetChain(ctx)
+	s.Require().NoError(err)
+
+	return chain
+}
+
+// CreateDANetwork creates a DA network using the provider
+func (s *DockerTestSuite) CreateDANetwork() tastoratypes.DataAvailabilityNetwork {
+	ctx := context.Background()
+
+	daNetwork, err := s.provider.GetDataAvailabilityNetwork(ctx)
+	s.Require().NoError(err)
+
+	return daNetwork
+}
+
+// CreateRollkitChain creates a Rollkit chain using the provider
+func (s *DockerTestSuite) CreateRollkitChain() tastoratypes.RollkitChain {
+	ctx := context.Background()
+
+	rollkitChain, err := s.provider.GetRollkitChain(ctx)
+	s.Require().NoError(err)
+
+	return rollkitChain
+}
+
+// StartBridgeNode initializes and starts a bridge node within the data availability network using the given parameters.
+func (s *DockerTestSuite) StartBridgeNode(ctx context.Context, bridgeNode tastoratypes.DANode, chainID string, genesisHash string, celestiaNodeHostname string) {
+	s.Require().Equal(tastoratypes.BridgeNode, bridgeNode.GetType())
+	err := bridgeNode.Start(ctx,
+		tastoratypes.WithChainID(chainID),
+		tastoratypes.WithAdditionalStartArguments("--p2p.network", chainID, "--core.ip", celestiaNodeHostname, "--rpc.addr", "0.0.0.0"),
+		tastoratypes.WithEnvironmentVariables(
+			map[string]string{
+				"CELESTIA_CUSTOM": tastoratypes.BuildCelestiaCustomEnvVar(chainID, genesisHash, ""),
+				"P2P_NETWORK":     chainID,
+			},
+		),
+	)
+	s.Require().NoError(err)
+}
+
+// FundWallet transfers the specified amount of utia from the faucet wallet to the target wallet.
+func (s *DockerTestSuite) FundWallet(ctx context.Context, wallet tastoratypes.Wallet, amount int64) {
+	fromAddress, err := sdkacc.AddressFromWallet(s.celestia.GetFaucetWallet())
+	s.Require().NoError(err)
+
+	toAddress, err := sdk.AccAddressFromBech32(wallet.GetFormattedAddress())
+	s.Require().NoError(err)
+
+	bankSend := banktypes.NewMsgSend(fromAddress, toAddress, sdk.NewCoins(sdk.NewCoin("utia", math.NewInt(amount))))
+	_, err = s.celestia.BroadcastMessages(ctx, s.celestia.GetFaucetWallet(), bankSend)
+	s.Require().NoError(err)
+}
+
+// StartRollkitNode initializes and starts a Rollkit node.
+func (s *DockerTestSuite) StartRollkitNode(ctx context.Context, bridgeNode tastoratypes.DANode, rollkitNode tastoratypes.RollkitNode) {
+	err := rollkitNode.Init(ctx)
+	s.Require().NoError(err)
+
+	bridgeNodeHostName, err := bridgeNode.GetInternalHostName()
+	s.Require().NoError(err)
+
+	authToken, err := bridgeNode.GetAuthToken()
+	s.Require().NoError(err)
+
+	daAddress := fmt.Sprintf("http://%s:26658", bridgeNodeHostName)
+	err = rollkitNode.Start(ctx,
```

Comment:
currently the support in tastora is limited to creation of rollkit nodes, and start/stop and exposing the hostname which is resolvable by the test runner. 

We can add issues as need functionality is required.

Reply:
Initial setup of evm stuff might take a while to figure out but I can look into it